### PR TITLE
LINK-1250 | Events without registration

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Modify registration parameter to return events with a registration when true, and without a registration when false

## Closes
[LINK-1250](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1250)

[LINK-1250]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1250?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: events/api.py [hunk 1]
```diff
@@ -2522,9 +2522,16 @@ def _filter_event_queryset(queryset, params, srs=None):  # noqa: C901
...
 
+    # filter to get events with a registration.
     val = params.get("registration", None)
     if val:
         queryset = queryset.exclude(registration=None)
+
+    # filter to get events without a registration.
+    val = params.get("registration!", None)
+    if val:
+        queryset = queryset.filter(registration=None)
```
Comment (charn) [inline events/api.py:2533]: It seems that using an exclamation mark is an existing pattern here, so I'm not going to say anything about that. Though I find it a bit misleading that you can get the same results/filtering with `registration=true` and `registration=false`,

Maybe the whole thing could be replaced with something like this?


```suggestion
    # filter to get events with or without a registration.
    val = params.get("registration", None)
    if val and parse_bool(val):
        queryset = queryset.exclude(registration=None)
    elif val: 
        queryset = queryset.filter(registration=None)
```

Comment (jorilindell) [reply events/api.py]: Very good idea. Changed as suggested, fixed the tests and updated the documentation

Comment (voneiden) [reply events/api.py]: Just a general note, this "do something if value is defined regardless of what the value is" elsewhere also an existing pattern linkedevents. But  :+1: for avoiding that.

**Review by voneiden — APPROVED** [review]
LGTM

**Pull request merged**